### PR TITLE
Upgrade to latest glob.

### PR DESCRIPTION
**Pull request by zzo**
Use sync glob (much faster) plus get stat info from glob itself to avoid having to stat to get mtime and if this is a directory or not

### CODE REVIEW

**Comment by zzo** [comment]
signed it

**Comment by zzo** [comment]
@Dignifiedquire what do you think of this?  Want to get this into the next canary and then put it all into the angular builds

File: lib/file_list.js [hunk 1]
```diff
@@ -71,6 +63,7 @@ var byPath = function (a, b) {
 var List = function (patterns, excludes, emitter, preprocess, batchInterval) {
   var self = this
   var pendingDeferred
+  var okToClearPendingDeferred = true
```
Comment (Dignifiedquire) [inline lib/file_list.js]: What exactly is this variable supposed to be tracking?

**Comment by Dignifiedquire** [comment]
Looks pretty good to me, that should make things faster :) 
The only thing I'm concerned about is windows..Have you tested this on windows machine if all the globbing is still working as expected there?

**Comment by zzo** [comment]
Agreed I'm also concerned about Windows. 
Because I replaced an asynchronous method (the original glob) and got rid of another ( the stat) it's now possible for 'complete' to be called before 'refresh' actually returns - which nulls out the promise that refresh returns. So in that case we need to hold off before null'ing it out. 
The main thing wrong here (well 'not ideal' let's say!) is the 'pending' file counting that's going on to determine when the promise should be fulfilled. Votja has a todo in there about getting rid of that. 
That was more refactoring than I was willing to stomach for this right now. There is also the issue of the same promise being reused by all of the other methods in this file which maybe makes sense and maybe it doesn't. Again I didn't want to completely rip apart this file although I'm sure it's ripe for it someday.

**Comment by Dignifiedquire** [comment]
There is also #1436 which is a bug in that file. I already promised to fix it, but that's only possible if I refactor this whole file..which is really badly needed.

**Comment by zzo** [comment]
Interesting. Multiple refreshes must happen all of the time wonder why this
doesn't happen more often.

**Comment by Dignifiedquire** [comment]
We don't want to push this into master, so closing this PR

**Pull request closed without merging**